libfreerdp/core/gateway/websocket.c

```diff
@@ -99,6 +100,41 @@ BOOL websocket_write_wstream(BIO* bio, wStream* sPacket, WEBSOCKET_OPCODE opcode
 	return TRUE;
 }
 
+static int websocket_write_all(BIO* bio, const BYTE* data, int length)
```

Comment:
Perhaps I'm wrong but this function looks a lot like `tls_write_all`, in fact it is very generic and has nothing to do with websockets (but IIRC that's also the case for `tls_write_all`).
Also don't we tend to use `size_t` for length parameters ?

Reply:
this is because it is the same function, but with BIO* instead of tls (since I don't have tls around here)
I can use size_t